src/job_cache/types.cpp

```diff
@@ -523,4 +523,27 @@ JAST FindJobRequest::to_json() const {
   return json;
 }
 
+FindJobResponse::FindJobResponse(JAST json) {
+  JAST found = json.get("found");
+  if (found.kind != JSON_TRUE) {
+    match = {};
+    return;
+  }
+  match = wcl::make_some<MatchingJob>(json.get("match"));
+}
+
+JAST FindJobResponse::to_json() const {
+  JAST json(JSON_OBJECT);
+
+  if (!match) {
+    json.add("found", static_cast<bool>(false));
```

Comment:
use `add_bool` instead of attempting the cast. I've found the cast to be unreliable and it caused me some headache.

Case and point there is no overload for bool alone so this was already clearly wrong. Also `false` is already a bool so static casting it was also wrong.